packages/tre/test/plugins/do/index.spec.ts

```diff
@@ -0,0 +1,160 @@
+import fs from "fs/promises";
+import path from "path";
+import { Miniflare, MiniflareOptions } from "@miniflare/tre";
+import test from "ava";
+import { getPort, useTmp } from "../../test-shared";
+
+const COUNTER_SCRIPT = (responsePrefix = "") => `export class Counter {
+  constructor(state) {
+    this.storage = state.storage;
+  }
+  async fetch(request) {
+    const count = ((await this.storage.get("count")) ?? 0) + 1;
+    void this.storage.put("count", count);
+    return new Response(${JSON.stringify(responsePrefix)} + count);
+  }
+}
+export default {
+  async fetch(request, env, ctx) {
+    const { pathname } = new URL(request.url);
+    const OBJECT = env[request.headers.get("MF-Test-Object") ?? "COUNTER"];
+    const id = OBJECT.idFromName(pathname);
+    const stub = OBJECT.get(id);
+    return stub.fetch(request);
+  },
+};`;
+
+test("persists Durable Object data in-memory between options reloads", async (t) => {
+  const opts: MiniflareOptions = {
+    port: await getPort(),
+    modules: true,
+    script: COUNTER_SCRIPT("Options #1: "),
+    durableObjects: { COUNTER: "Counter" },
+  };
+  const mf = new Miniflare(opts);
+  t.teardown(() => mf.dispose());
+
+  let res = await mf.dispatchFetch("http://localhost");
+  t.is(await res.text(), "Options #1: 1");
+
+  opts.script = COUNTER_SCRIPT("Options #2: ");
+  await mf.setOptions(opts);
+  res = await mf.dispatchFetch("http://localhost");
+  t.is(await res.text(), "Options #2: 2");
+
+  opts.durableObjectsPersist = false;
+  opts.script = COUNTER_SCRIPT("Options #3: ");
+  await mf.setOptions(opts);
+  res = await mf.dispatchFetch("http://localhost");
+  t.is(await res.text(), "Options #3: 3");
+
+  opts.durableObjectsPersist = "memory:";
+  opts.script = COUNTER_SCRIPT("Options #4: ");
+  await mf.setOptions(opts);
+  res = await mf.dispatchFetch("http://localhost");
+  t.is(await res.text(), "Options #4: 4");
+});
+
+test("persists Durable Object data on file-system", async (t) => {
+  const tmp = await useTmp(t);
+  const opts: MiniflareOptions = {
+    name: "worker",
+    port: await getPort(),
+    modules: true,
+    script: COUNTER_SCRIPT(),
+    durableObjects: { COUNTER: "Counter" },
+    durableObjectsPersist: tmp,
+  };
+  const mf = new Miniflare(opts);
+  t.teardown(() => mf.dispose());
+
+  let res = await mf.dispatchFetch("http://localhost");
+  t.is(await res.text(), "1");
+
+  // Check directory created for "worker"'s Durable Object
+  const names = await fs.readdir(tmp);
+  t.deepEqual(names, ["worker-Counter"]);
+
+  // Check reloading keeps persisted data
```

Comment:
It would be good to see a test that kept data across a restart, not just a reload